src/server/game/Entities/Player/Player.cpp

```diff
@@ -30836,3 +30836,44 @@ bool Player::CanExecutePendingSpellCastRequest()
 
     return true;
 }
+
+void Player::AddMoveImpulse(Position direction)
+{
+    auto addImpulse = WorldPackets::Movement::MoveAddImpulse();
+    addImpulse.MoverGUID = GetGUID();
+    addImpulse.SequenceIndex = m_movementCounter++;
+    addImpulse.Direction = direction;
+    SendMessageToSet(addImpulse.Write(), true);
+}
+
+void Player::InitAdvancedFly()
+{
+    FlightCapabilityEntry const* flightCapabilityEntry = sFlightCapabilityStore.LookupEntry(GetFlightCapabilityID());
+    if (!flightCapabilityEntry)
+        return;
+
+    std::vector<std::tuple<OpcodeServer, float, Optional<float>>> advFlyValues = {
```

Comment:
personally i wouldn't use a container here but construct each packet individually, as constructing the container and deconstructing it is time we could save at this point. We could still apply functional decomposition like you did here though.

something along
```c
        auto advFlyingPacket = WorldPackets::Movement::SetAdvFlyingSpeed(SMSG_MOVE_SET_ADV_FLYING_BANKING_RATE);
        advFlyingPacket.SequenceIndex = m_movementCounter++;
        advFlyingPacket.speed = flightCapabilityEntry->BankingRateMin;
        advFlyingPacket.maxSpeed = flightCapabilityEntry->BankingRateMax;
        SendDirectMessage(advFlyingPacket.Write());
```
we could even go a step further and put everything in constructor, like
```c
SendDirectMessage(WorldPackets::Movement::SetAdvFlyingSpeed(SMSG_MOVE_SET_ADV_FLYING_BANKING_RATE, flightCapabilityEntry->BankingRateMin, flightCapabilityEntry->BankingRateMax, _movementCounter++).Write());
```
that way we'd have even less code in this method - and its way more performant

(also on a side note of using containers here: std::vector would be overkill here aswell, an array would be enough)

Reply:
Done, i still used an proxy method because i'd rather have a single point to modify if the packet change at some point in the future, and it should have no impact on perfs with inline